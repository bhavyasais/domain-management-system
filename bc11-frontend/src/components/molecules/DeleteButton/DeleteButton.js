import React from "react";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import deleteSelectedIcon from '../../../assets/delete.png';
import IconAtom from '../../atoms/Icon/IconAtom'
const useStyles = makeStyles(() => ({
    button: {
        width: "169px",
        height: "40px",
    },
}));

const DeleteButton = ({ deleteSelected }) => {
    const classes = useStyles();
    const theme = useTheme();
    return (
        <ButtonAtom
            variant="contained"
            color={theme.palette.error.main}
            className={classes.button}
            startIcon={<IconAtom icon={<img src={deleteSelectedIcon} width="20" height="20"/>} />}
            value={"Delete selected("+`${deleteSelected}`+")"}
            textVariant="caption"
        />
    );
}
export default DeleteButton;