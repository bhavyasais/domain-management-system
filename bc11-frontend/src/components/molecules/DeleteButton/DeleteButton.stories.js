import React from "react";
import DeleteButton from "./DeleteButton";

export default {
  title: "Molecules/DeleteButton",
};

export const deleteButton = () => {
  return <DeleteButton deleteSelected="3"/>;
};
