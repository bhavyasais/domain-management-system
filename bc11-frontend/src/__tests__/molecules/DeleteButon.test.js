import React from "react";
import renderer from "react-test-renderer";
import DeleteButton from "../../components/molecules/DeleteButton/DeleteButton";

describe("DeleteButton", () => {
    it("should render correctly", () => {
        let wrapper;
        wrapper = renderer.create(<DeleteButton variant="contained" value="Delete selected" />);
        expect(wrapper).toMatchSnapshot();
    });
});
