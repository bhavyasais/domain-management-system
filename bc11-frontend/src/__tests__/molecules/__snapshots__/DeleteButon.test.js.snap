// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton should render correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#f44336",
        "maxWidth": undefined,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label makeStyles-buttonLabel-2"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <div>
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root"
            style={
              Object {
                "color": undefined,
              }
            }
          >
            <img
              height="20"
              src="delete.png"
              width="20"
            />
          </span>
        </div>
      </span>
      <span
        className="MuiTypography-root makeStyles-buttonLabel-2 MuiTypography-caption"
      >
        Delete selected(undefined)
      </span>
    </span>
  </button>
</div>
`;
